### ISSUE TITLE
ability to use own icons and colour scheme , like a mod for amidst

### ISSUE THREAD

**Issue by BlueSteelAUS** [toolbox4minecraft/amidst#107]
ability to use own icons and colour scheme , like a mod for amidst

**Comment by Treer** [comment]
You can [customise the colours](https://github.com/toolbox4minecraft/amidst/wiki/Biome-Profiles)

**Comment by BlueSteelAUS** [comment]
Sorry i thought that was for the highlight colours. anyway still would like to mod the icons for the mines,villiges etc..

**Comment by stefandollase** [comment]
You can always replace the icons in the jar file:

1. open the Amidst jar file with any zip program
2. go to `/amidst/gui/main/icon/`
3. replace any icon that you don't like

Amidst will use these icons when it is started the next time. Please let me know whether that works for you.

**Comment by BlueSteelAUS** [comment]
I did it i've changed the stronghold and nether icons [https://i.imgsafe.org/84dedd2.png](url) if you'd like to use them i can share them I think they show up much better

**Comment by stefandollase** [comment]
I am not sure about the icons. I kind of like them. However, the lava does not seem to indicate a Nether Fortress very well. I would like to hear the opinion of @Treer about this, since he also made many of the icons. I am also easy to impress when it comes to graphics, so he is probably more competent than I am when it comes to graphics.

Also, before we can use your icons, I need to make sure that you agree to the statement about the [Intellectual Property Rights of Contributions](https://github.com/toolbox4minecraft/amidst/wiki/Supporting-the-Development#intellectual-property-rights-of-contributions).

**Comment by BlueSteelAUS** [comment]
the hing is i chose lava as its abundant in the nether and looks very different to the other icons, to me at the moment the nether icon is way to dark and hard to spot. you could use a blaze icon if don't like lava but again i think lava suits the nether. The strong hold icon was way to close to the dungeon icons. they just look grey to me and have to look extremely hard to see wither its a stronghold or dungeon, I chose the ender eye because its what most people want strongholds foe to go the end

**Comment by BlueSteelAUS** [comment]
I created / edited the icons myself, (i'm not an artist, and took me ages, I used gimp)

**Comment by BigAlanM** [comment]
Sort of off topic:
but I like the idea of making the Stronghold icon a brighter color so that they can be more easily seen.
The same is true for Nether Fortresses.
It wan't much of a problem with only three as you could always click on "Goto Stronghold" and get a list of coordinates but now with 128 strongholds in 1.9 that is no longer reasonable when displaying those maps.
I've taken to repeatedly hitting Ctrl-3 (for Strongholds) or Ctrl-8 (for Nether Fortresses) to make the icons blink so I can find them on the map.

**Comment by BlueSteelAUS** [comment]
You shouldn't have to :) that shows that they aren't the right icons. lol

**Comment by Treer** [comment]
I like that Ender eye icon for the Strongholds, here's a variant: ![stronghold](https://cloud.githubusercontent.com/assets/6390507/13457130/9780bece-e0ba-11e5-8785-6e1e455a5036.png)  ![stronghold3](https://cloud.githubusercontent.com/assets/6390507/13457395/2261d680-e0bc-11e5-8814-67a918d59fec.png)

Not suggesting the variant is better - just exploring options. If we change the Stronghold icon, should we change the Amidst icon? (It has that icon because it started life as a stronghold finder)

I agree the Nether Fortress icon is bad, but I don't know if the lava is much better. Perhaps Nether Wort?

**Comment by BlueSteelAUS** [comment]
it does need something bright. for the nether blaze or Nether Wart, i do prefer your Ender eyes to mine. lol , like i said I'm no artist. lol

**Comment by stefandollase** [comment]
Out of the three, I like @Treer's left stronghold icon the best, which is the one with the bigger eye.

Regarding the Nether Fortress, I think the Nether Wart is a great idea. It is easy to recognized and clearly stands for the Fortress. However, I am a little bit worried that it might also be too dark. I guess we have to try this out.

I have mixed feelings about switching the Amidst Icon. On the one hand, Amidst has been more than a simple stronghold finder for quite some time. This did also not lead to a new icon. I think the Amidst icon is now just the Amidst icon and not necessarily related to strongholds. Thus, I don't think there is a need to change it when the stronghold icon is changed. On the other hand, the current icon is really simple and kind of boring, so if you come up with an interesting new icon we might as well replace it. Again, I don't think the new icon has to be related to strongholds.

**Comment by BlueSteelAUS** [comment]
[new nether fortress icon sample](https://i.imgsafe.org/3b25fd7.png)
Note that it's an isometric tile with a gold background to stand out from the rest of the map
 
new screenshot of amidst using Treer's ender eys and the newest suggested nether fortress icon
[Screenshot](https://i.imgsafe.org/3fa8fad.png)

**Comment by BigAlanM** [comment]
I want to warn anyone that changes the icons that if you ever plan on posting your Amidst maps on forums or a blog etc. you're going to have to also tell people what those, non-standard, icons represent    so that you don't confuse your audience.
I've been doing that already for the new maps I've posted that show the new Witch Hut icon.
I figure the Igloo Icon is self explanatory.
Amidst maps have been posted on the Minecraft forums from the time it first came out. People expect them to look a certain way.
This advice also applicable to using non-standard biome colors.  I warn people when I post a modified map.

**Comment by BlueSteelAUS** [comment]
agreed .. customised icons/colour screens should be listed in posts on forums when you use your own colour screen. and also for the first couple of months if the default ones change on the released version

**Comment by stefandollase** [comment]
I don't like the new Nether Fortress icon. I cannot recognise it as a Nether Wart. Maybe, something between the second and third stage in [this image](http://www.minecraft101.net/reference/images/nether-wart.png) would work out?

@BigAlanM You are right, we should be careful to change well known icons. However, do you think we should not change the icons at all, even if they are better recognisable? It will be easier for people how see an Amidst map for the first time to know that the icons stand for.

**Comment by BlueSteelAUS** [comment]
ok i'll try to come up with something better, the colours of nether wart are dark that's why i placed them onto soul sand as an isometric tile . with the bright yellow background to see the icon better on the map
i'll try with your images and see what i can come up with

**Comment by BlueSteelAUS** [comment]
hows this
 ![nether fortress](https://i.imgsafe.org/3d43792.png)
 ![amidst screenshot](https://i.imgsafe.org/ce73696.png)

**Comment by stefandollase** [comment]
This is already better. However, it is still very dark. Did you try to make the background transparent?

**Comment by BlueSteelAUS** [comment]
the thing is transparent BG doesn't work well on all colours.
![nether fortress icon](https://i.imgsafe.org/7286d0f.png)
have a look at the ones in the ocean compared to the others
![Screenshot](https://i.imgsafe.org/20a625b.png)

**Comment by BlueSteelAUS** [comment]
ok i think i've got something here :)
i've added a slight lightning tint to the transparency, you can now still see whats under it like mineshafts and slime chunks. its no longer too dark, and you can see in easily in all biomes
![nf3](https://i.imgsafe.org/d0aead9.png)
![ss2](https://i.imgsafe.org/9ae2fa0.png)

**Comment by Treer** [comment]
![nether_fortress2](https://cloud.githubusercontent.com/assets/6390507/13719840/48d0733a-e850-11e5-8cea-553f60e1c1b2.png)
![nether_fortress3](https://cloud.githubusercontent.com/assets/6390507/13719841/4c7b7d0e-e850-11e5-9458-4c9eee108b2c.png)

**Comment by BlueSteelAUS** [comment]
Your top icon look ok to me to me at least its still better than the original dark brown tile

I rather like the idea of semi Transparent BG, I know the nether is dark, maybe Treer's top icon with a filters transparence as the BG to replace the nether brick..

**Comment by Treer** [comment]
What's the purpose of using a semi-transparent background? It's a deviation from the other solid icons — tho that doesn't necessarily mean it's wrong. Aesthetic?

**Comment by BlueSteelAUS** [comment]
if you look at the one i did you can see in the screenshot, you can see mineshafts etc showing through the semi transparent areas while only lightening the map underneath a little. for viewing some of the possibly stacked icons and terrain that's actually under it

**Comment by stefandollase** [comment]
I like @Treer's first icon. Since it is lightened, it is probably easy to see on the map. Did you try to remove the nether-brick background and to draw the strong black line directly around the nether wart?
